packages/babel-helper-member-expression-to-functions/src/index.js

```diff
@@ -337,6 +337,8 @@ const handle = {
     // [MEMBER = _VALUE] = ARR -> [_destructureSet(MEMBER) = _VALUE] = ARR
     // [...MEMBER] -> [..._destructureSet(MEMBER)]
     if (
+      (parentPath.isForXStatement() &&
+        parentPath.parentPath.isAssignmentPattern({ left: node })) ||
```

Comment:
If a `path` is a `ForXStatement`, how can its `parentPath` be an `AssignmentPattern`? Note that the children of an `Pattern` is always an expression. A statement can not be an expression.

Here is an AST example of `for (o.p of arr);`
```json
// for (o.p of arr);
{
  "type": "ForOfStatement",
  "await": false,
  "left": {
    "type": "MemberExpression",
    "object": { "type": "Identifier", "name": "o" },
    "property": { "type": "Identifier", "name": "p" }
   },
  "right": { "type": "Identifier", "name": "arr" }
}
```

The `parentPath` here refers to the `NodePath` of `ForOfStatement`. `NodePath` is a wrapper of the AST node that provides `parent` association with other `NodePath`. In this PR We would like to make sure that `member`, which refers to `MemberExpression` here, is indeed the `left` of the underlying AST node of `parentPath`.

So we can use
```js
parentPath.isForXStatement() && parentPath.node.left === node
```

The query function `isForXStatement` accepts an additional parameter as query, so the code above can be simplified as

```js
parentPath.isForXStatement({ left: node })
```

Reply:
ok, when I run the test I seem to get the same results, which shouldn't happen?

**input**
```
class D {
  #arr;
  f() {
    for (const el of this.#arr);
  }
}
```

**output**
```
var _arr = new WeakMap();

var D = /*#__PURE__*/function () {
  "use strict";

  function D() {
    babelHelpers.classCallCheck(this, D);

    _arr.set(this, {
      writable: true,
      value: void 0
    });
  }

  babelHelpers.createClass(D, [{
    key: "f",
    value: function f() {
      for (var el of babelHelpers.classPrivateFieldGet(this, _arr)) {
        ;
      }
    }
  }]);
  return D;
}();
```

Reply:
The output is expected. Can you add a test case from the original issue?
```js
class D {
  #arr;
  f() {
    for (const this.#arr of [1, 2]);
  }
}
```